src/lib/registration/ui/not_installed_products_dialog.rb

```diff
@@ -0,0 +1,225 @@
+        #   not installed. (1/2)
+        summary = _("<p>The addons listed below are registered but not installed: </p>")
+
+        summary << "<ul>#{not_installed_addon_names.map { |a| "<li>#{a}</li>" }.join("")}</ul>"
+
+        # TRANSLATORS: A RichText warning about all the products registered but
+        #   not installed. (2/2)
+        summary << _("<p><b>Synchronize</b> your products if you want to <b>deactivate</b> " \
+                     "them at your registration server.</p>")
+
+        summary
+      end
+
+      # It updates the summary of registered but not installed products
+      # updating also Addon cache.
+      def update_summary
+        log.info "Updating summary"
+        Yast::Popup.Feedback(RegistrationUI::CONTACTING_MESSAGE, _("Updating summary")) do
+          Addon.reset!
+          Addon.find_all(registration)
+        end
+
+        Yast::UI.ChangeWidget(Id(:summary), :Value, not_installed_addons_summary)
+      end
+
+      # @return [Boolean] true if in the registration server exists some
+      # registered but not installed addon; false otherwise
+      def registered_not_installed_addons?
+        !Addon.registered_not_installed.empty?
+      end
+
+      # @return [Array<String>] return an array with all the registered but not
+      #   installed addon names.
+      def not_installed_addon_names
+        Addon.registered_not_installed.map(&:name)
+      end
+
+      # @param addon [Registration::Addon] addon to be installed
+      # @return [Boolean] true if the given addon is installable
+      def addon_product_installable?(addon)
+        product = product_from_addon_repos(addon)
+        return false if !product || !Yast::Pkg.ResolvableInstall(product["name"], :product)
+
+        if !Yast::Pkg.PkgSolve(true)
+          return false if Yast::PackagesUI.RunPackageSelector("mode" => :summaryMode) != :accept
+        end
+
+        Yast::PackagesUI.ConfirmLicenses
+      end
+
+      # Find the product resolvables matching the repository url with the addon
+      # ones.
+      #
+      # @param addon [Registration::Addon]
+      # @return [Hash] product which repository url match with addon ones
+      def product_from_addon_repos(addon)
+        Yast::Pkg.ResolvableProperties("", :product, "").find do |product|
+          return false if product["status"] != :available
+
+          product_url = SwMgmt.repository_data(product["source"]).fetch("url", "")
+
+          addon.repositories.any? { |r| no_params_url(product_url) == no_params_url(r["url"]) }
+        end
+      end
+
+      # Given an URL return the concatenation of host and path removing all the
+      # end slashes.
+      #
+      # FIXME: This method could be moved to UrlHelpers or to URL module.
+      #
+      # @param [String] url to parse
+      # @return [String] url without params
+      def no_params_url(url)
+        uri = URI.parse(url)
+        "#{uri.host}#{uri.path.gsub(/\/+$/, "")}"
```

Comment:
I'd rather compare also the other URL properties, it's unlikely that the same URL just with a different scheme (or user/password) will be in the system, but it's better to be on the safe side:

```ruby
uri.query = nil
uri.path.gsub!(/\/+\z/, "")
```

Then you can compare the URI objects directly.